adapters/huaweiads/huaweiads.go

```diff
@@ -0,0 +1,1320 @@
+		dspImpTrackings2StrImg.WriteString("<img height=\"1\" width=\"1\" src='" + dspImpTrackings[i] + "' >  ")
+	}
+
+	adm = "<style> html, body  " +
+		"{ margin: 0; padding: 0; width: 100%; height: 100%; vertical-align: middle; }  " +
+		"html  " +
+		"{ display: table; }  " +
+		"body { display: table-cell; vertical-align: middle; text-align: center; -webkit-text-size-adjust: none; }  " +
+		"</style> " +
+		"<span class=\"title-link advertiser_label\">" + imageTitle + "</span> " +
+		"<a href='" + clickUrl + "' style=\"text-decoration:none\" " +
+		"onclick=sendGetReq()> " +
+		"<img src='" + imageInfoUrl + "' width='" + strconv.Itoa(int(adWidth)) + "' height='" + strconv.Itoa(int(adHeight)) + "'/> " +
+		"</a> " +
+		dspImpTrackings2StrImg.String() +
+		"<script type=\"text/javascript\">" +
+		"var dspClickTrackings = [" + dspClickTrackings + "];" +
+		"function sendGetReq() {" +
+		"sendSomeGetReq(dspClickTrackings)" +
+		"}" +
+		"function sendOneGetReq(url) {" +
+		"var req = new XMLHttpRequest();" +
+		"req.open('GET', url, true);" +
+		"req.send(null);" +
+		"}" +
+		"function sendSomeGetReq(urls) {" +
+		"for (var i = 0; i < urls.length; i++) {" +
+		"sendOneGetReq(urls[i]);" +
+		"}" +
+		"}" +
+		"</script>"
+	return adm, adWidth, adHeight, nil
+}
+
+func getDspImpClickTrackings(content *content) (dspImpTrackings []string, dspClickTrackings string) {
+	if content.Monitor != nil {
+		for _, monitor := range content.Monitor {
+			if monitor.EventType == "imp" && len(monitor.Url) != 0 {
+				dspImpTrackings = monitor.Url
+			}
+			if monitor.EventType == "click" && len(monitor.Url) != 0 {
+				dspClickTrackings = getStrings(monitor.Url)
+			}
+		}
+	}
+	return dspImpTrackings, dspClickTrackings
+}
+
+// Add url for monitor
+func getStrings(eles []string) string {
+	if len(eles) == 0 {
+		return ""
+	}
+	var strs strings.Builder
+	for i := 0; i < len(eles); i++ {
+		strs.WriteString("\"" + eles[i] + "\"")
+		if i < len(eles)-1 {
+			strs.WriteString(",")
+		}
+	}
+	return strs.String()
+}
+
+// get duration, millisecond -> format: 00:00:00.000
+func getDuration(duration int64) string {
+	// duration millisecond
+	if duration == 0 {
+		return "00:00:00.000"
+	}
+	var totalSec = int(duration) / 1000
+	var mmm = int(duration) % 1000
+	var hour = totalSec / 3600
+	var tmp = totalSec % 3600
+	var min = tmp / 60
+	var sec = tmp % 60
+	var result string
+	result += addZero(strconv.Itoa(hour), ":", false)
+	result += addZero(strconv.Itoa(min), ":", false)
+	result += addZero(strconv.Itoa(sec), ".", false)
+	result += addZero(strconv.Itoa(mmm), "", true)
+	return result
+}
+
+func addZero(str string, dot string, isMmm bool) (result string) {
+	if isMmm == false {
+		if len(str) < 2 {
+			result = "0" + str + dot
+		} else {
+			result = str + dot
+		}
+	} else {
+		if len(str) == 1 {
+			result = "00" + str
+		} else if len(str) == 2 {
+			result = "0" + str
+		} else {
+			result = str
+		}
+	}
+	return result
```

Comment:
Since the if/else structure is exhaustive, you can make return from each line, ex:
```
if isMmm == false {
	if len(str) < 2 {
		return "0" + str + dot
	} else {
		return str + dot
	}
} else {
	if len(str) == 1 {
		return "00" + str
	} else if len(str) == 2 {
		return "0" + str
	} else {
		return str
	}
}
```

Reply:
done